src/Installer/PluginInstaller.php

```diff
@@ -2,89 +2,235 @@
+            foreach ($dir as $info) {
+                if (!$info->isDir() || $info->isDot()) {
+                    continue;
+                }
+
+                $name = $info->getFilename();
+                $plugins[$name] = $pluginsDir . DIRECTORY_SEPARATOR . $name;
+            }
+        }
+
+        ksort($plugins);
+        return $plugins;
+    }
+
+    /**
+     * Rewrite the config file with a complete list of plugins
      *
-     * @throws \InvalidArgumentException if $initial package is not installed
+     * @param string $configFile the path to the config file
+     * @param array $plugins of plugins
+     * @return void
      */
-    public function update(InstalledRepositoryInterface $repo, PackageInterface $initial, PackageInterface $target)
+    public static function writeConfigFile($configFile, $plugins)
     {
-        parent::update($repo, $initial, $target);
+        $root = dirname(dirname($configFile));
 
-        $ns = $this->primaryNamespace($initial);
-        $this->updateConfig($ns, null);
+        $data = [];
+        foreach ($plugins as $name => $pluginPath) {
+            $pluginPath = str_replace(
+                DIRECTORY_SEPARATOR . DIRECTORY_SEPARATOR,
+                DIRECTORY_SEPARATOR,
+                $pluginPath
+            );
 
-        $path = $this->getInstallPath($target);
-        $ns = $this->primaryNamespace($target);
-        $this->updateConfig($ns, $path);
+            // Normalize to *nix paths.
+            $pluginPath = str_replace('\\', '/', $pluginPath);
+            $pluginPath .= '/';
+
+            $data[] = sprintf("        '%s' => '%s'", $name, $pluginPath);
+        }
+
+        $data = implode(",\n", $data);
+
+        $contents = <<<PHP
+<?php
+\$baseDir = dirname(dirname(__FILE__));
+return [
+    'plugins' => [
+$data
+    ]
+];
+
+PHP;
+
+        $root = str_replace(
+            DIRECTORY_SEPARATOR . DIRECTORY_SEPARATOR,
+            DIRECTORY_SEPARATOR,
+            $root
+        );
+
+        // Normalize to *nix paths.
+        $root = str_replace('\\', '/', $root);
+        $contents = str_replace('\'' . $root, '$baseDir . \'', $contents);
+        file_put_contents($configFile, $contents);
     }
 
     /**
-     * Uninstalls specific package.
+     * Path to the plugin config file
      *
-     * @param \Composer\Repository\InstalledRepositoryInterface $repo Repository in which to check.
-     * @param \Composer\Package\PackageInterface $package Package instance.
+     * @return string absolute file path
      */
-    public function uninstall(InstalledRepositoryInterface $repo, PackageInterface $package)
+    protected static function _configFile($root)
     {
-        parent::uninstall($repo, $package);
-        $path = $this->getInstallPath($package);
-        $ns = $this->primaryNamespace($package);
-        $this->updateConfig($ns, null);
+        return $root . DIRECTORY_SEPARATOR . 'vendor' . DIRECTORY_SEPARATOR . 'cakephp-plugins.php';
```

Comment:
Use: `$this->composer->getConfig()->get('vendor-dir');` instead of hardcoded 'vendor'